### PR TITLE
Adds Widget walkthrough

### PR DESCRIPTION
**Pull request by jmelberg-okta**
## Description:
- Adds quickstart walkthrough for Okta Sign-In Widget

### Resolves:

* [OKTA-138200](https://oktainc.atlassian.net/browse/OKTA-138200)

### CODE REVIEW

File: _source/quickstart/widget/default-example.md [hunk 1]
```diff
@@ -1,6 +1,103 @@
...
-{% include quickstart-coming-soon.html %}
\ No newline at end of file
+This guide will walk you through integrating the [Okta Sign-In Widget](https://github.com/okta/okta-signin-widget) into an app by performing these steps:
+
+1. Add an OpenID Connect Client in Okta
+2. Add Sign-In Widget assets to your project
+3. Implement Okta Sign-In
+4. Handle the Authorization Code Flow
+
+At the end of the Sign-In Widget instructions you can choose your server type to learn more about post-authentication workflows, such as exchanging the `authorization code` for tokens that your server can use to communicate with other servers.
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: > tokens that your server can use to communicate with other servers

Would a better example be "tokens that your server can use to get information about the user"?

Comment (jmelberg-okta) [inline _source/quickstart/widget/default-example.md]: Yep. I'm actually in the process of updating this to use the implicit flow instead.. so this section will change.

Thanks for taking a look though!

**Comment by nbarbettini** [comment]
@jmelberg-okta How come we're changing it to the implicit flow? This is for a server-rendered app use case, right?

File: _source/quickstart/widget/default-example.md [hunk 2]
```diff
@@ -1,6 +1,103 @@
...
+            { el: '#okta-login-container' },
+            function (res) {
+                if (res.status !== 'SUCCESS') {
+                   return;
+                }
+                // Success!
+            }
+        );
+</script>
+```
```
Comment (jmelberg-okta) [inline _source/quickstart/widget/default-example.md]: Created a `error` callback to with a comment on handling exceptions.

File: _source/quickstart/widget/default-example.md [hunk 3]
```diff
@@ -1,6 +1,135 @@
...
+
+## Add an OpenID Connect Client
+* Log into the Okta Developer Dashboard, click **Applications** then **Add Application**.
+* Choose **Single Page App (SPA)** as the platform, then populate your new OpenID Connect application with values similar to:
+
+| Setting             | Value                                                 |
+| ------------------- | ----------------------------------------------------- |
+| Application Name    | My Web App                                            |
+| Base URIs           | http://localhost:{port}                               |
+| Login redirect URIs | http://localhost:{port}/callback                      |
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: The Add Application wizard defaults to `http://localhost:3000/implicit/callback` - any specific reason why we're changing that here?

File: _source/quickstart/widget/default-example.md [hunk 4]
```diff
@@ -1,6 +1,135 @@
...
+href="https://ok1static.oktacdn.com/assets/js/sdk/okta-signin-widget/2.1.0/css/okta-theme.css"
+type="text/css"
+rel="stylesheet"/>
+```
+
+> The `okta-sign-in.min.js` file will expose a global `OktaSignIn` object that can bootstrap the widget.
+
+## Implement Okta Sign-In
+
+You can render the widget anywhere on the page by creating a `<div>` with a unique `id.`  You can also control the visual look of the widget by adding your own CSS.
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: nit: move the period outside of `id.`

File: _source/quickstart/widget/default-example.md [hunk 5]
```diff
@@ -1,6 +1,135 @@
...
+        clientId: "{clientId}",
+        authParams: {
+            issuer: "https://{yourOktaDomain}.com/oauth2/default",
+            responseType: 'token id_token',
+            responseMode: 'query',
+            scopes: ['openid', 'profile', 'email']
+        }
+        });
+
+        // Render the widget to the CSS selector okta-login-container
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: nit: the selector is `#okta-login-container`

File: _source/quickstart/widget/default-example.md [hunk 6]
```diff
@@ -1,6 +1,135 @@
...
+                }
+                // Store the tokens in the TokenManager in the order requestsed
+                oktaSignIn.tokenManager.add('accessToken', res[0])
+                oktaSignIn.tokenManager.add('idToken', res[1])
+                
+                return;
+            },
+            function error(err) {
+                // The widget handles most types of errors: CONFIG_ERROR, OAUTH_ERROR, etc
+                // Add any custom logic to handle uncaught exceptions
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: Suggestion: add a dummy `console.log(err)` statement to give the dev an idea of what "handle uncaught exceptions" might look like.

File: _source/quickstart/widget/default-example.md [hunk 7]
```diff
@@ -1,6 +1,135 @@
...
+        });
+
+        // Render the widget to the CSS selector okta-login-container
+        oktaSignIn.renderEl(
+            { el: '#okta-login-container' },
+            function success(res) {
+                if (res.status !== 'SUCCESS') {
+                   return;
+                }
+                // Store the tokens in the TokenManager in the order requestsed
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: nit: requestsed -> requested

File: _source/quickstart/widget/default-example.md [hunk 8]
```diff
@@ -1,6 +1,135 @@
...
+                // The widget handles most types of errors: CONFIG_ERROR, OAUTH_ERROR, etc
+                // Add any custom logic to handle uncaught exceptions
+            }
+        );
+</script>
+```
+
+### Using the Access Token
+
+Your application now has an access token in local storage that was issued by your Okta Authorization server.  You can read this token and present it to your own server to authenticate requests for resources on your server.  As a hypothetical example, let's say that you have an API that gives us messages for our user.  You could create a `callMessagesApi` function that requires authentication, retrieves the access token from local storage, and attach it to our resource request.
```
Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: Suggestion: "You can use this token to authenticate requests for resources on your server or API."

Comment (nbarbettini) [inline _source/quickstart/widget/default-example.md]: nit: the grammar sounds odd at the end of the sentence. Suggestion: "You could create a `callMessagesApi` function that gets the access token from local storage and uses it to make an authenticated request to your server."

File: _source/quickstart/widget/default-example.md (hunk 3 above)
Comment (jmelberg-okta) [inline _source/quickstart/widget/default-example.md]: This was an oversight on my end. Good catch!

File: _source/quickstart/widget/default-example.md [hunk 9]
```diff
@@ -1,6 +1,136 @@
...
+    
+    if (!accessToken) {
+        return;
+    }
+
+    // Make the request using jQuery
+    $.ajax({
+        url: '/api/messages',
+        headers: {
+        Authorization : 'Bearer ' + accessToken
```
Comment (lboyette-okta) [inline _source/quickstart/widget/default-example.md]: strange indentation

File: _source/quickstart/widget/default-example.md [hunk 10]
```diff
@@ -1,6 +1,136 @@
...
+
+    var oktaSignIn = new OktaSignIn({
+        baseUrl: "https://{yourOktaDomain}.com",
+        clientId: "{clientId}",
+        authParams: {
+            issuer: "https://{yourOktaDomain}.com/oauth2/default",
+            responseType: 'token id_token',
+            responseMode: 'query',
+            scopes: ['openid', 'profile', 'email']
+        }
```
Comment (lboyette-okta) [inline _source/quickstart/widget/default-example.md]: strange indentation

**Pull request merged**